### PR TITLE
Proposed -02 version

### PR DESCRIPTION
**Pull request by smyslov**

### CODE REVIEW

File: draft-ietf-ipsecme-ikev2-multiple-ke-00.xml [hunk 1]
```diff
@@ -513,11 +519,27 @@ Initiator                             Responder
...
+
                 <t> The responder performs negotiation using standard IKEv2 procedure described in Section 3.3 of <xref target="RFC7296"/>.
                 However, for the Additional Key Exchange types the responder's choice MUST NOT contain equal transform IDs (apart from NONE), 
                 and the ID selected for Transform Type 4 MUST NOT appear in any of Additional Key Exchange transforms.
                 In other words, all selected key exchange methods must be different.
                 </t>
+
+                <t> If the responder selected NONE for some Additional Key Exchange types (provided they were proposed by the initiator), 
+                then the corresponding IKE_INTERMEDIATE exchanges should not take place. The IKE_INTERMEDIATE exchanges
+                MUST only be performed for Additional Key Exchange types cintaining non-NONE responders choices.
```
Comment (ctjhai) [inline draft-ietf-ipsecme-ikev2-multiple-ke-00.xml]:
```suggestion
                MUST only be performed for Additional Key Exchange types containing non-NONE responders choices.
```
```suggestion
                MUST only be performed for Additional Key Exchange types containing non-NONE responders choices.
```

File: draft-ietf-ipsecme-ikev2-multiple-ke-00.xml [hunk 2]
```diff
@@ -393,14 +400,14 @@
...
+            Transform Types are defined, each sharing allowed Transform IDs 
             with Transform Type 4.  The IKE_SA_INIT message includes one or more newly defined SA transforms 
 			that lists the extra key exchange policy required by the
 			initiator; the responder selects a single transform of each type, and
 			returns them in the response IKE_SA_INIT message. 
             Then, provided that additional key exchanges are negotiated, the initiator and the responder 
-            perform one or more IKE_INTERMEDIATE exchanges; each such exchange includes a KE payload 
-            for one of the negotiated key exchanges.</t>
+            perform one or more IKE_INTERMEDIATE exchanges; every such exchange includes a KE payload 
+            for the next method from the negotiated list of them.</t>
```
Comment (ctjhai) [inline draft-ietf-ipsecme-ikev2-multiple-ke-00.xml]:
```suggestion
            for the next method from the negotiated list.</t>
```
```suggestion
            for the next method from the negotiated list.</t>
```

**Review by ctjhai — APPROVED** [review]

**Review by sfluhrer — APPROVED** [review]

File: draft-ietf-ipsecme-ikev2-multiple-ke-00.xml (hunk 1 above)
Comment (smyslov) [reply draft-ietf-ipsecme-ikev2-multiple-ke-00.xml]: I agree, thank you for catching this typo.

File: draft-ietf-ipsecme-ikev2-multiple-ke-00.xml (hunk 2 above)
Comment (smyslov) [reply draft-ietf-ipsecme-ikev2-multiple-ke-00.xml]: Agree.

**Pull request merged**